lib/fa2/nft/nft.impl.jsligo

```diff
@@ -1,282 +1,70 @@
-         Tezos.transaction(Main(callback_param), 0mutez, callback);
-      return [list([operation]), s]
-   };
-
-   export const update_operators = <T>(updates: TZIP12.update_operators, s: storage<T>): ret<T> => {
-      const update_operator = (
-         [operators, update]: [operators, TZIP12.unit_update]
-      ): operators =>
-         match(update) {
-            when (Add_operator(operator)):
-               add_operator(
-                  operators,
-                  operator.owner,
-                  operator.operator,
-                  operator.token_id
-               )
-            when (Remove_operator(operator)):
-               remove_operator(
-                  operators,
-                  operator.owner,
-                  operator.operator,
-                  operator.token_id
-               )
-         };
-      let operators = get_operators(s);
-      operators = List.fold_left(update_operator, operators, updates);
-      const store = set_operators([s, operators]);
-      return [list([]), store]
-   };
-
-   export const get_balance = <T>(p: [address, nat], s: storage<T>): nat => {
-      const [owner, token_id] = p;
-      Assertions.assert_token_exist(s.token_metadata, token_id);
-      if (is_owner_of(s.ledger, token_id, owner)) {
-         return 1n
-      } else {
-         return 0n
-      }
-   };
-
-   export const total_supply = <T>(token_id: nat, s: storage<T>): nat => {
-      Assertions.assert_token_exist(s.token_metadata, token_id);
-      return 1n
-   };
-
-   export const all_tokens = <T>(_: unit, _s: storage<T>): set<nat> =>
-      failwith(Errors.not_available);
-
-   export const is_operator = <T>(op: TZIP12.operator, s: storage<T>): bool => {
-      const authorized =
-         match(Big_map.find_opt([op.owner, op.operator], s.operators)) {
-            when (Some(a)):
-               a
-            when (None()):
-               Set.empty
-         };
-      return (Set.mem(op.token_id, authorized) || op.owner == op.operator)
-   };
-
-   export const token_metadata = <T>(p: nat, s: storage<T>): TZIP12.tokenMetadataData => {
-      return match(Big_map.find_opt(p, s.token_metadata)) {
-         when (Some(data)):
-            data
-         when (None()):
-            failwith(Errors.undefined_token)
-      }
-   }
+#import "./extendable_nft.impl.jsligo" "NFTExtendable"
+
+export type ledger = NFTExtendable.ledger;
+export type operator = NFTExtendable.operator;
+export type operators = NFTExtendable.operators;
+export type storage = {
+  ledger : ledger,
+  operators : operators,
+  token_metadata : TZIP12.tokenMetadata,
+  metadata : TZIP16.metadata,
+}
+type ret = [list<operation>, storage];
+
+@inline
+const lift = (s: storage): NFTExtendable.storage<unit> => {
+  return {
+    extension: unit,
+    ledger: s.ledger,
+    operators: s.operators,
+    token_metadata: s.token_metadata,
+    metadata: s.metadata
+  };
 }
 
-export namespace NFT implements TZIP12Interface.FA2 {
```

Comment:
No more constraint between implementation and his interface ? I understand it improve the API

Reply:
It's true that by removing some modules like that we have to drop some explicit constraints on the interfaces of modules. I think we could either add a module instantiation somewhere which enforces this constraint (`module Dummy : TZIP12Interface.FA2 = NFT`), as it was done in the past (but I removed it in the previous PR), or just consider that the library's users can always use these interfaces themselves. What do you prefer?

Reply:
I think I prefer to enforce contraint (code is living and add constraint make it stronger in case of update) + it's better for understanding . But I think it's not possible to do it via jsligo ? 
So
> Consider that the library's users can always use these interfaces themselves
looks to be good for now